### PR TITLE
Add support to exclude architectures and repositories

### PR DESCRIPTION
**Pull request by wlozano0collabora**
In some use cases it is useful to exclude architectures and repositories as those builds should not block a pipeline. To allow that add a set of new arguments to dput.

### CODE REVIEW

File: obo-core/src/actions.rs [hunk 1]
```diff
@@ -72,6 +72,10 @@ pub struct DputAction {
     pub rebuild_if_unchanged: bool,
     #[clap(long, default_value = "")]
     pub message: String,
+    #[clap(long)]
+    pub excluded_arch: Vec<String>,
```
Comment (refi64) [inline obo-core/src/actions.rs]: These should probably be `exclude_` since the other arguments are all imperative (`branch-to`, `rebuild-if`, etc).

Also the new options should be mentioned in the top-level README.

File: obo-core/src/actions.rs [hunk 2]
```diff
@@ -264,6 +270,8 @@ impl Actions {
                     disabled_repos: DisabledRepos::Skip {
                         wait_options: Default::default(),
                     },
+                    excluded_arch: args.excluded_arch.clone(),
```
Comment (refi64) [inline obo-core/src/actions.rs]: I don't think the `.clone`s are necessary in this second branch, since it's the last usage in the function? (But I'm not 100% sure on that.)

File: obo-core/src/build_meta.rs [hunk 3]
```diff
@@ -57,6 +57,19 @@ pub enum DisabledRepos {
 pub struct BuildMetaOptions {
     pub history_retrieval: BuildHistoryRetrieval,
     pub disabled_repos: DisabledRepos,
+    pub excluded_arch: Vec<String>,
+    pub excluded_repo: Vec<String>,
+}
+
+impl Default for BuildMetaOptions {
```
Comment (refi64) [inline obo-core/src/build_meta.rs]: If this is for the tests only, I'd rather they either use some small helper function (e.g. `fn make_options(BuildHistoryRetrieval, DisabledRepos)`) or just pass the two parameters each time, instead of implementing Default for everyone, which the caller probably *doesn't want to actually use*.

File: obo-core/src/build_meta.rs
Comment (refi64) [inline obo-core/src/build_meta.rs:366]: After this, both architectures are added, so it would be a great place to add a test for `excluded_arches`. We should also test `excluded_repos`, but that might be a bit more annoying since none of the current setup utilizes multiple repos at all; maybe stick it at the very end?

File: obo-core/src/actions.rs (hunk 1 above)
Comment (wlozano0collabora) [reply obo-core/src/actions.rs]: Sure.

File: obo-core/src/build_meta.rs (hunk 3 above)
Comment (wlozano0collabora) [reply obo-core/src/build_meta.rs]: Yes, it was just to make the test pass. Let me tweak it.

File: obo-core/src/actions.rs (hunk 1 above)
Comment (wlozano0collabora) [reply obo-core/src/actions.rs]: Implemented!

File: obo-core/src/build_meta.rs (hunk 3 above)
Comment (wlozano0collabora) [reply obo-core/src/build_meta.rs]: Done.

File: obo-core/src/actions.rs (hunk 2 above)
Comment (wlozano0collabora) [reply obo-core/src/actions.rs]: You are right, fixed. Thank you!

File: obo-core/src/build_meta.rs
Comment (wlozano0collabora) [reply obo-core/src/build_meta.rs:366]: Done.

File: obo-core/src/actions.rs [hunk 4]
```diff
@@ -264,6 +270,8 @@ impl Actions {
                     disabled_repos: DisabledRepos::Skip {
                         wait_options: Default::default(),
                     },
+                    exclude_arch: args.exclude_arch,
+                    exclude_repo: args.exclude_repo,
```
[the remark this answers is not in the document]
Comment (wlozano0collabora) [reply obo-core/src/actions.rs:274]: @refi64 this is due to the change you suggested, which made sense to me but copilot seems to be not so happy, with? Do you agree in making copilot happy?

[the remark this answers is not in the document]
Comment (refi64) [reply obo-core/src/actions.rs:274]: imo not worth "fixing" something where copilot is just wrong
